### PR TITLE
drivers/audio: fix samp rate conversion issue

### PR DESCRIPTION
**Pull request by jinxiuxu**
## Summary
use ac_controls.hw to get the correct sample rate intead of ac_controls.b

### CODE REVIEW

**Comment by acassis** [comment]
Please check CI error

File: drivers/audio/es8311.c [hunk 1]
```diff
@@ -865,11 +863,11 @@ static int es8311_getcaps(FAR struct audio_lowerhalf_s *dev, int type,
           {
               /* Report supported input sample rates */
 
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr =
+              caps->ac_controls.hw[0] =
                 AUDIO_SAMP_RATE_11K | AUDIO_SAMP_RATE_16K |
                 AUDIO_SAMP_RATE_22K | AUDIO_SAMP_RATE_32K |
                 AUDIO_SAMP_RATE_44K | AUDIO_SAMP_RATE_48K;
+              caps->ac_controls.hw[1] = 0;
```
Comment (xiaoxiang781216) [inline drivers/audio/es8311.c:870]: why add

File: drivers/audio/es8311.c [hunk 2]
```diff
@@ -847,11 +845,11 @@ static int es8311_getcaps(FAR struct audio_lowerhalf_s *dev, int type,
 
               /* 8kHz is hardware dependent */
 
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr =
+              caps->ac_controls.hw[0] =
                 AUDIO_SAMP_RATE_11K | AUDIO_SAMP_RATE_16K |
                 AUDIO_SAMP_RATE_22K | AUDIO_SAMP_RATE_32K |
                 AUDIO_SAMP_RATE_44K | AUDIO_SAMP_RATE_48K;
+              caps->ac_controls.hw[1] = 0;
```
Comment (xiaoxiang781216) [inline drivers/audio/es8311.c:852]: ditto

File: drivers/audio/vs1053.c [hunk 3]
```diff
@@ -649,14 +647,13 @@ static int vs1053_getcaps(FAR struct audio_lowerhalf_s *lower, int type,
...
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr = AUDIO_SAMP_RATE_8K  |
-                     AUDIO_SAMP_RATE_11K |
-                     AUDIO_SAMP_RATE_16K |
-                     AUDIO_SAMP_RATE_22K |
-                     AUDIO_SAMP_RATE_32K |
-                     AUDIO_SAMP_RATE_44K |
-                     AUDIO_SAMP_RATE_48K;
+              caps->ac_controls.hw[0] = AUDIO_SAMP_RATE_8K  |
+                                       AUDIO_SAMP_RATE_11K |
```
Comment (xiaoxiang781216) [inline drivers/audio/vs1053.c:651]: align

File: arch/arm/src/cxd56xx/cxd56_nxaudio.c [hunk 4]
```diff
@@ -2633,8 +2631,8 @@ static int cxd56_getcaps(struct audio_lowerhalf_s *lower, int type,
 
               /* Report supported output sample rates */
 
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr = CXD56_SUPP_RATES_L;
+              caps->ac_controls.hw[0] = CXD56_SUPP_RATES_L;
+              caps->ac_controls.hw[1] = CXD56_SUPP_RATES_H;
```
Comment (xiaoxiang781216) [inline arch/arm/src/cxd56xx/cxd56_nxaudio.c:2635]: CXD56_SUPP_RATES_L and CXD56_SUPP_RATES_H into CXD56_SUPP_RATES

File: arch/arm/src/cxd56xx/cxd56_nxaudio.c [hunk 5]
```diff
@@ -2654,8 +2652,8 @@ static int cxd56_getcaps(struct audio_lowerhalf_s *lower, int type,
 
               /* Report supported input sample rates */
 
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr = CXD56_SUPP_RATES_L;
+              caps->ac_controls.hw[0] = CXD56_SUPP_RATES_L;
+              caps->ac_controls.hw[1] = CXD56_SUPP_RATES_H;
```
Comment (xiaoxiang781216) [inline arch/arm/src/cxd56xx/cxd56_nxaudio.c:2656]: ditto

File: drivers/audio/es8311.c (hunk 1 above)
Comment (jinxiuxu) [reply drivers/audio/es8311.c:870]: Previously removed when adding 12K support, now added back.

File: drivers/audio/es8311.c (hunk 2 above)
Comment (jinxiuxu) [reply drivers/audio/es8311.c:852]: ditto

File: arch/arm/src/cxd56xx/cxd56_nxaudio.c [hunk 6]
```diff
@@ -2633,8 +2631,7 @@ static int cxd56_getcaps(struct audio_lowerhalf_s *lower, int type,
 
               /* Report supported output sample rates */
 
-              ptr  = (uint16_t *)caps->ac_controls.b;
-              *ptr = CXD56_SUPP_RATES_L;
+              caps->ac_controls.hw[0] = CXD56_SUPP_RATES;
```
Comment (xiaoxiang781216) [inline arch/arm/src/cxd56xx/cxd56_nxaudio.c:2634]: please update line 83-96

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**